### PR TITLE
Update to Upstream V0.27.0

### PR DESCRIPTION
**Pull request by HashEngineering**

### CODE REVIEW

File: contrib/test.sh [hunk 1]
```diff
@@ -1,25 +1,69 @@
...
+  cargo build --verbose --features="no-std $FEATURES" --no-default-features
+
+  # Build specific features
+  for feature in ${FEATURES}
+  do
+      cargo build --verbose --features="no-std $feature"
+  done
+
+  cargo run --example bip32 L1HKVVLHXiUhecWnwFYF6L3shkf1E12HUmuZTESvBXUdx3yqVP1D
+  cargo run --no-default-features --features no-std --example bip32 L1HKVVLHXiUhecWnwFYF6L3shkf1E12HUmuZTESvBXUdx3yqVP1D
```
Comment (gruve-p) [inline contrib/test.sh]: These WIF addresses needs to be fixed

File: embedded/README.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+source ./scripts/env.sh && cargo run --target thumbv7m-none-eabi
+```
+
+Output should be something like:
+
+```text
+heap size 524288
+secp buf size 66240
+Seed WIF: L1HKVVLHXiUhecWnwFYF6L3shkf1E12HUmuZTESvBXUdx3yqVP1D
+Address: bc1qpx9t9pzzl4qsydmhyt6ctrxxjd4ep549np9993
```
Comment (gruve-p) [inline embedded/README.md]: Wif address and address needs fixing

File: embedded/src/main.rs [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+fn main() -> ! {
+    hprintln!("heap size {}", HEAP_SIZE).unwrap();
+
+    unsafe { ALLOCATOR.init(cortex_m_rt::heap_start() as usize, HEAP_SIZE) }
+
+    let size = Secp256k1::preallocate_size();
+    hprintln!("secp buf size {}", size*16).unwrap();
+
+    // Load a private key
+    let raw = "L1HKVVLHXiUhecWnwFYF6L3shkf1E12HUmuZTESvBXUdx3yqVP1D";
```
Comment (gruve-p) [inline embedded/src/main.rs]: WIF address needs fixing

File: embedded/src/main.rs [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+
+    let mut buf_ful = vec![AlignedType::zeroed(); size];
+    let secp = Secp256k1::preallocated_new(&mut buf_ful).unwrap();
+
+    // Derive address
+    let pubkey = pk.public_key(&secp);
+    let address = Address::p2wpkh(&pubkey, Network::Groestlcoin).unwrap();
+    hprintln!("Address: {}", address).unwrap();
+
+    assert_eq!(address.to_string(), "bc1qpx9t9pzzl4qsydmhyt6ctrxxjd4ep549np9993".to_string());
```
Comment (gruve-p) [inline embedded/src/main.rs]: Address needs fixing

File: examples/bip32.rs [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+use groestlcoin::secp256k1::ffi::types::AlignedType;
+
+fn main() {
+    // This example derives root xprv
+    // from a 32-byte secret of the input WIF string,
+    // derives the child xprv with path m/84h/17h/0h,
+    // prints out corresponding xpub,
+    // calculates and prints out the first receiving segwit address.
+    // Run this example with cargo and WIF argument:
+    // cargo run --example bip32 L1HKVVLHXiUhecWnwFYF6L3shkf1E12HUmuZTESvBXUdx3yqVP1D
```
Comment (gruve-p) [inline examples/bip32.rs]: Example WIF needs fixing

File: src/util/address.rs [hunk 6]
```diff
@@ -653,32 +691,78 @@ mod tests {
...
-            ("bc1pw508d6qejxtdg4y5r3zarvary0c5xw7kw508d6qejxtdg4y5r3zarvary0c5xw7k7grplx", "5128751e76e8199196d454941c45d1b3a323f1433bd6751e76e8199196d454941c45d1b3a323f1433bd6"),
-            ("BC1SW50QA3JX3S", "6002751e"),
-            ("bc1zw508d6qejxtdg4y5r3zarvaryvg6kdaj", "5210751e76e8199196d454941c45d1b3a323"),
+            ("bc1pw508d6qejxtdg4y5r3zarvary0c5xw7kw508d6qejxtdg4y5r3zarvary0c5xw7kt5nd6y", "5128751e76e8199196d454941c45d1b3a323f1433bd6751e76e8199196d454941c45d1b3a323f1433bd6"),
+            ("BC1SW50QGDZ25J", "6002751e"),
+            ("bc1zw508d6qejxtdg4y5r3zarvaryvaxxpcs", "5210751e76e8199196d454941c45d1b3a323"),
             ("tb1qqqqqp399et2xygdj5xreqhjjvcmzhxw4aywxecjdzew6hylgvsesrxh6hy", "0020000000c4a5cad46221b2a187905e5266362b99d5e91c6ce24d165dab93e86433"),
+            ("tb1pqqqqp399et2xygdj5xreqhjjvcmzhxw4aywxecjdzew6hylgvsesf3hn0c", "5120000000c4a5cad46221b2a187905e5266362b99d5e91c6ce24d165dab93e86433"),
+            ("bc1p0xlxvlhemja6c4dqv22uapctqupfhlxm9h8z3k2e72q4k9hcz7vqzk5jj0", "512079be667ef9dcbbac55a06295ce870b07029bfcdb2dce28d959f2815b16f81798")
```
Comment (gruve-p) [inline src/util/address.rs]: these are bitcoin addresses

File: src/util/address.rs [hunk 7]
```diff
@@ -653,32 +691,78 @@ mod tests {
...
+            // Invalid program length for witness version 0 (per BIP141)
+            "BC1QR508D6QEJXTDG4Y5R3ZARVARYV98GJ9P",
+            // Mixed case
+            "tb1p0xlxvlhemja6c4dqv22uapctqupfhlxm9h8z3k2e72q4k9hcz7vq47Zagq",
+            // zero padding of more than 4 bits
+            "bc1p0xlxvlhemja6c4dqv22uapctqupfhlxm9h8z3k2e72q4k9hcz7v07qwwzcrf",
+            // Non-zero padding in 8-to-5 conversion
+            "tb1p0xlxvlhemja6c4dqv22uapctqupfhlxm9h8z3k2e72q4k9hcz7vpggkg4j",
+            // Empty data section
+            "bc1gmk9yu",
```
Comment (gruve-p) [inline src/util/address.rs:739]: these are bitcoin addresses

File: src/util/address.rs [hunk 8]
```diff
@@ -653,32 +691,78 @@ mod tests {
...
             "bc1zw508d6qejxtdg4y5r3zarvaryvqyzf3du",
+            // Non-zero padding in 8-to-5 conversion
             "tb1qrp33g0q5c5txsp9arysrx4k6zdkfs4nce4xj0gdcccefvpysxf3pjxtptv",
-            "bc1gmk9yu",
+            // Final test for empty data section is the same as above in BIP-350
+
+            // 3. BIP-173 valid test vectors obsolete by BIP-350
+            "bc1pw508d6qejxtdg4y5r3zarvary0c5xw7kw508d6qejxtdg4y5r3zarvary0c5xw7k7grplx",
+            "BC1SW50QA3JX3S",
+            "bc1zw508d6qejxtdg4y5r3zarvaryvg6kdaj",
```
Comment (gruve-p) [inline src/util/address.rs:765]: these are bitcoin addresses

File: src/util/address.rs [hunk 9]
```diff
@@ -746,4 +830,18 @@ mod tests {
             hex_script!("001454d26dddb59c7073c6a197946ea1841951fa7a74")
         );
     }
+
+    #[test]
+    fn test_qr_string() {
+        for el in  ["132F25rTsvBdp9JzLLBHP5mvGY66i1xdiM", "33iFwdLuRpW1uK1RTRqsoi8rR4NpDzk66k"].iter() {
```
Comment (gruve-p) [inline src/util/address.rs]: these are bitcoin addresses

File: src/util/base58.rs [hunk 10]
```diff
@@ -267,15 +272,16 @@ mod tests {
...
+        let exp = "2hXiLBbH9kmBoA9HWRRpWuHu29Hg49ApfMGzw6CXJkHn2sVUFjm8qF8FQQm29jFNXHjUXs59Q\
+        K9gk2MrWjiLokRHQ2t71bY5HjQUmutR2rTU91sc9gZjixxegH45Zs19PkCweG4bshVEm3XvqSDnELis8yjxkb\
+        aeV3DFCuYA69mxwtsqZpJxqWCSbu9vPBAoX5bWKeF5bRXNSd9tp4F7GMWCmvQqC2fxpVqk78YkxCs6H4EAJncp\
+        sNxAxCwrLQ164Pzn2y5zupcmLmwEf1SRmN997q5J3rU3ytRf3ot1iuymf";
         assert_eq!(&res, exp);
 
         // Addresses
-        let addr = hex_decode("24F8917303BFA8EF24F292E8FA1419B20460BA064D").unwrap();
+        let addr = Vec::from_hex("00f8917303bfa8ef24f292e8fa1419b20460ba064d").unwrap();
         assert_eq!(&check_encode_slice(&addr[..]), "1PfJpZsjreyVrqeoAfabrRwwjQyoSQMmHH");
```
Comment (gruve-p) [inline src/util/base58.rs]: `1PfJpZsjreyVrqeoAfabrRwwjQyoSQMmHH` is a bitcoin address

File: src/util/ecdsa.rs [hunk 11]
```diff
@@ -0,0 +1,568 @@
...
+        assert_eq!(&sk.to_string(), "cVt4o7BGAig1UXywgGSmARhxMdzP5qvQsxKkSsc1XEkw3tDTQFpy");
+        let sk_str =
+            PrivateKey::from_str("cVt4o7BGAig1UXywgGSmARhxMdzP5qvQsxKkSsc1XEkw3tDTQFpy").unwrap();
+        assert_eq!(&sk.to_wif(), &sk_str.to_wif());
+
+        // mainnet uncompressed
+        let sk = PrivateKey::from_wif("5JYkZjmN7PVMjJUfJWfRFwtuXTGB439XV6faajeHPAM9Z2PT2R3").unwrap();
+        assert_eq!(sk.network, Groestlcoin);
+        assert_eq!(sk.compressed, false);
+        assert_eq!(&sk.to_wif(), "5JYkZjmN7PVMjJUfJWfRFwtuXTGB439XV6faajeHPAM9Z2PT2R3");
```
Comment (gruve-p) [inline src/util/ecdsa.rs]: These are either bitcoin wif keys or bitcoin addresses

File: src/util/ecdsa.rs [hunk 12]
```diff
@@ -0,0 +1,568 @@
...
+
+        let secp = Secp256k1::new();
+        let mut pk = sk.public_key(&secp);
+        assert_eq!(pk.compressed, false);
+        assert_eq!(&pk.to_string(), "042e58afe51f9ed8ad3cc7897f634d881fdbe49a81564629ded8156bebd2ffd1af191923a2964c177f5b5923ae500fca49e99492d534aa3759d6b25a8bc971b133");
+        assert_eq!(pk, PublicKey::from_str("042e58afe51f9ed8ad3cc7897f634d881fdbe49a81564629ded8156bebd2ffd1af191923a2964c177f5b5923ae500fca49e99492d534aa3759d6b25a8bc971b133").unwrap());
+        let addr = Address::p2pkh(&pk, sk.network);
+        assert_eq!(&addr.to_string(), "1GhQvF6dL8xa6wBxLnWmHcQsurx9RxiMc8");
```
Comment (gruve-p) [inline src/util/ecdsa.rs]: this is a bitcoin address

File: src/util/ecdsa.rs [hunk 13]
```diff
@@ -0,0 +1,568 @@
...
+        assert_eq!(pk.wpubkey_hash().unwrap().to_hex(), "9511aa27ef39bbfa4e4f3dd15f4d66ea57f475b4");
+        assert_eq!(upk.wpubkey_hash(), None);
+    }
+
+    #[cfg(feature = "serde")]
+    #[test]
+    fn test_key_serde() {
+        use serde_test::{Configure, Token, assert_tokens};
+
+        static KEY_WIF: &'static str = "cVt4o7BGAig1UXywgGSmARhxMdzP5qvQsxKkSsc1XEkw3tDTQFpy";
```
Comment (gruve-p) [inline src/util/ecdsa.rs]: bitcoin private key

**Review by gruve-p — CHANGES_REQUESTED** [review]
running 169 tests
test blockdata::block::tests::block_test ... FAILED
test blockdata::block::tests::block_version_test ... ok
test blockdata::block::tests::compact_roundrtip_test ... ok
test blockdata::block::tests::segwit_block_test ... FAILED
test blockdata::block::tests::test_coinbase_and_bip34 ... ok
test blockdata::block::tests::validate_pow_test ... FAILED
test blockdata::constants::test::bitcoin_genesis_first_transaction ... ok
test blockdata::constants::test::bitcoin_genesis_full_block ... FAILED
test blockdata::constants::test::signet_genesis_full_block ... ok
test blockdata::constants::test::testnet_genesis_full_block ... FAILED
test blockdata::opcodes::tests::str_roundtrip ... ok
test blockdata::script::test::defult_dust_value_tests ... ok
test blockdata::script::test::op_return_test ... ok
test blockdata::script::test::p2sh_p2wsh_conversion ... ok
test blockdata::script::test::provably_unspendable_test ... ok
test blockdata::script::test::script ... ok
test blockdata::script::test::script_asm ... ok
test blockdata::script::test::script_builder ... ok
test blockdata::script::test::script_builder_verify ... ok
test blockdata::script::test::script_generators ... ok
test blockdata::script::test::script_hashes ... ok
test blockdata::script::test::script_ord ... ok
test blockdata::script::test::script_p2pk ... ok
test blockdata::script::test::script_p2sh_p2p2k_template ... ok
test blockdata::script::test::script_serialize ... ok
test blockdata::script::test::scriptint_round_trip ... ok
test blockdata::script::test::test_iterator ... ok
test blockdata::transaction::tests::test_is_coinbase ... ok
test blockdata::transaction::tests::test_nonsegwit_transaction ... ok
test blockdata::transaction::tests::test_ntxid ... ok
test blockdata::transaction::tests::test_outpoint ... ok
test blockdata::transaction::tests::test_segwit_transaction ... ok
test blockdata::transaction::tests::test_sighash ... FAILED
test blockdata::transaction::tests::test_sighashtype_fromstr_display ... ok
test blockdata::transaction::tests::test_sighashtype_standard ... ok
test blockdata::transaction::tests::test_transaction_version ... ok
test blockdata::transaction::tests::test_txid ... ok
test blockdata::transaction::tests::test_txin ... ok
test blockdata::transaction::tests::test_txin_default ... ok
test blockdata::transaction::tests::tx_no_input_deserialization ... ok
test consensus::encode::tests::deserialize_checkeddata_test ... ok
test consensus::encode::tests::deserialize_int_test ... ok
test consensus::encode::tests::deserialize_nonminimal_vec ... ok
test consensus::encode::tests::deserialize_strbuf_test ... ok
test consensus::encode::tests::deserialize_vec_test ... ok
test consensus::encode::tests::limit_read_test ... ok
test consensus::encode::tests::serialization_round_trips ... ok
test consensus::encode::tests::serialize_checkeddata_test ... ok
test consensus::encode::tests::serialize_int_test ... ok
test consensus::encode::tests::serialize_strbuf_test ... ok
test consensus::encode::tests::serialize_varint_test ... ok
test consensus::encode::tests::serialize_vector_test ... ok
test network::address::test::addrv2message_test ... ok
test network::address::test::debug_format_test ... ok
test network::address::test::deserialize_address_test ... ok
test network::address::test::deserialize_addrv2_test ... ok
test network::address::test::onion_test ... ok
test network::address::test::serialize_address_test ... ok
test network::address::test::serialize_addrv2_test ... ok
test network::address::test::test_socket_addr ... ok
test network::constants::tests::serialize_test ... ok
test network::constants::tests::service_flags_test ... ok
test network::constants::tests::string_test ... ok
test network::message::test::commandstring_test ... ok
test network::message::test::deserialize_getaddr_test ... FAILED
test network::message::test::deserialize_partial_message_test ... FAILED
test network::message::test::deserialize_version_test ... FAILED
test network::message::test::full_round_ser_der_raw_network_message_test ... ok
test network::message::test::serialize_getaddr_test ... ok
test network::message::test::serialize_mempool_test ... ok
test network::message::test::serialize_ping_test ... ok
test network::message::test::serialize_verack_test ... ok
test network::message_blockdata::tests::getblocks_message_test ... ok
test network::message_blockdata::tests::getheaders_message_test ... ok
test network::message_network::tests::version_message_test ... ok
test network::stream_reader::test::parse_multipartmsg_test ... FAILED
test network::stream_reader::test::read_block_from_file_test ... ok
test network::stream_reader::test::read_doublemsgs_test ... FAILED
test network::stream_reader::test::read_multipartmsg_test ... FAILED
test network::stream_reader::test::read_sequencemsg_test ... FAILED
test network::stream_reader::test::read_singlemsg_test ... FAILED
test util::address::tests::test_bip173_350_vectors ... FAILED
test util::address::tests::test_non_existent_segwit_version ... ok
test util::address::tests::test_p2pkh_address_58 ... ok
test util::address::tests::test_p2pkh_from_key ... ok
test util::address::tests::test_p2sh_address_58 ... ok
test util::address::tests::test_p2sh_parse ... ok
test util::address::tests::test_p2shwpkh ... ok
test util::address::tests::test_p2shwsh ... ok
test util::address::tests::test_p2wpkh ... ok
test util::address::tests::test_p2wsh ... ok
test util::address::tests::test_qr_string ... FAILED
test util::amount::tests::add_sub_mul_div ... ok
test util::amount::tests::checked_arithmetic ... ok
test util::amount::tests::floating_point ... ok
test util::amount::tests::from_str ... ok
test util::amount::tests::parsing ... ok
test util::amount::tests::test_overflows ... ok
test util::amount::tests::test_unsigned_signed_conversion ... ok
test util::amount::tests::to_from_string_in ... ok
test util::amount::tests::to_string ... ok
test util::amount::tests::to_string_with_denomination_from_str_roundtrip ... ok
test util::base58::tests::test_base58_decode ... ok
test util::base58::tests::test_base58_encode ... FAILED
test util::base58::tests::test_base58_roundtrip ... ok
test util::bip143::tests::bip143_p2wpkh ... ok
test util::bip143::tests::bip143_p2wpkh_nested_in_p2sh ... ok
test util::bip143::tests::bip143_p2wsh_nested_in_p2sh ... ok
test util::bip143::tests::bip143_sighash_flags ... ok
test util::bip158::test::test_bit_stream ... ok
test util::bip158::test::test_blockfilters ... FAILED
test util::bip158::test::test_filter ... ok
test util::bip32::tests::fmt_child_number ... ok
test util::bip32::tests::test_derivation_path_conversion_index ... ok
test util::bip32::tests::test_increment ... ok
test util::bip32::tests::test_parse_derivation_path ... ok
test util::bip32::tests::test_vector_1 ... ok
test util::bip32::tests::test_vector_2 ... ok
test util::bip32::tests::test_vector_3 ... ok
test util::contracthash::tests::bad_key_number ... ok
test util::contracthash::tests::sanity ... ok
test util::contracthash::tests::script ... ok
test util::contracthash::tests::tweak_fixed_vector ... ok
test util::contracthash::tests::tweak_secret ... ok
test util::ecdsa::tests::pubkey_read_write ... ok
test util::ecdsa::tests::test_key_derivation ... FAILED
test util::ecdsa::tests::test_pubkey_hash ... ok
test util::ecdsa::tests::test_wpubkey_hash ... ok
test util::endian::tests::endian_chunk_test ... ok
test util::endian::tests::endianness_test ... ok
test util::merkleblock::tests::merkleblock_construct_from_txids_found ... FAILED
test util::merkleblock::tests::merkleblock_construct_from_txids_not_found ... FAILED
test util::merkleblock::tests::merkleblock_serialization ... ok
test util::merkleblock::tests::pmt_malleability ... ok
test util::merkleblock::tests::pmt_tests ... ok
test util::misc::tests::test_script_codesep_remove ... ok
test util::misc::tests::test_script_find_and_remove ... ok
test util::misc::tests::test_signed_msg_hash ... ok
test util::psbt::tests::bip_vectors::invalid_vector_1 - should panic ... ok
test util::psbt::tests::bip_vectors::invalid_vector_2 - should panic ... ok
test util::psbt::tests::bip_vectors::invalid_vector_3 - should panic ... ok
test util::psbt::tests::bip_vectors::invalid_vector_4 - should panic ... ok
test util::psbt::tests::bip_vectors::invalid_vector_5 - should panic ... ok
test util::psbt::tests::bip_vectors::valid_vector_1 ... ok
test util::psbt::tests::bip_vectors::valid_vector_2 ... ok
test util::psbt::tests::bip_vectors::valid_vector_3 ... FAILED
test util::psbt::tests::bip_vectors::valid_vector_4 ... ok
test util::psbt::tests::bip_vectors::valid_vector_5 ... ok
test util::psbt::tests::bip_vectors::valid_vector_6 ... ok
test util::psbt::tests::deserialize_and_serialize_psbt_with_two_partial_sigs ... ok
test util::psbt::tests::serialize_and_deserialize_preimage_psbt ... ok
test util::psbt::tests::serialize_and_deserialize_proprietary ... ok
test util::psbt::tests::serialize_then_deserialize_global ... ok
test util::psbt::tests::serialize_then_deserialize_output ... ok
test util::psbt::tests::serialize_then_deserialize_psbtkvpair ... ok
test util::psbt::tests::trivial_psbt ... ok
test util::taproot::test::test_midstates ... ok
test util::taproot::test::test_vectors_core ... ok
test util::uint::tests::mul_u32_test ... ok
test util::uint::tests::multiplication_test ... ok
test util::uint::tests::uint256_arithmetic_test ... ok
test util::uint::tests::uint256_bits_test ... ok
test util::uint::tests::uint256_bitslice_test ... ok
test util::uint::tests::uint256_comp_test ... ok
test util::uint::tests::uint256_display_test ... ok
test util::uint::tests::uint256_extreme_bitshift_test ... ok
test util::uint::tests::uint256_serialize_test ... ok
test util::uint::tests::uint_from_be_bytes ... ok
test util::uint::tests::uint_to_be_bytes ... ok

failures:

---- blockdata::block::tests::block_test stdout ----
thread 'main' panicked at 'assertion failed: `(left == right)`
  left: `4c917a410f4e899195f816081844e56aceda71c4cc4fe634aebe9437e57344bf`,
 right: `b41742d162370e98c9f2ff03f3d169bc56d0ac76c1d37f782b047ea00ca57a9e`', src/blockdata/block.rs:364:9
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

---- blockdata::block::tests::segwit_block_test stdout ----
thread 'main' panicked at 'assertion failed: `(left == right)`
  left: `7ef6e8a89489bf99fc1b53552c00a6408bc2d03d15a620d42a672f0ae726bc10`,
 right: `32047c1fa6b1f44b770e8f5c4d33761b7fb3f4a535f7d4b91d675445516a92fb`', src/blockdata/block.rs:399:9

---- blockdata::block::tests::validate_pow_test stdout ----
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: BlockBadProofOfWork', src/blockdata/block.rs:435:68

---- blockdata::constants::test::bitcoin_genesis_full_block stdout ----
thread 'main' panicked at 'assertion failed: `(left == right)`
  left: `504365055`,
 right: `486604799`', src/blockdata/constants.rs:207:9

---- blockdata::constants::test::testnet_genesis_full_block stdout ----
thread 'main' panicked at 'assertion failed: `(left == right)`
  left: `503382015`,
 right: `486604799`', src/blockdata/constants.rs:221:9

---- blockdata::transaction::tests::test_sighash stdout ----
thread 'main' panicked at 'assertion failed: `(left == right)`
  left: `53730274f133dd1ce2bba16b97322ac2530a051c94f1ea1d20b6e9ea05ca670f`,
 right: `7e3197893b2cb5da782b138d07ba0eeb4c70314daa5c87f6d5f9f36c7a16af31`', src/blockdata/transaction.rs:1008:9

---- network::message::test::deserialize_getaddr_test stdout ----
thread 'main' panicked at 'assertion failed: msg.is_ok()', src/network/message.rs:525:9

---- network::message::test::deserialize_partial_message_test stdout ----
thread 'main' panicked at 'assertion failed: msg.is_ok()', src/network/message.rs:586:9

---- network::message::test::deserialize_version_test stdout ----
thread 'main' panicked at 'assertion failed: msg.is_ok()', src/network/message.rs:551:9

---- network::stream_reader::test::parse_multipartmsg_test stdout ----
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: InvalidChecksum { expected: [43, 63, 135, 243], actual: [27, 249, 170, 234] }', src/network/stream_reader.rs:206:42

---- network::stream_reader::test::read_doublemsgs_test stdout ----
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: InvalidChecksum { expected: [55, 14, 242, 191], actual: [190, 97, 184, 39] }', src/network/stream_reader.rs:226:42

---- network::stream_reader::test::read_multipartmsg_test stdout ----
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: InvalidChecksum { expected: [55, 14, 242, 191], actual: [190, 97, 184, 39] }', src/network/stream_reader.rs:280:42

---- network::stream_reader::test::read_sequencemsg_test stdout ----
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: InvalidChecksum { expected: [55, 14, 242, 191], actual: [190, 97, 184, 39] }', src/network/stream_reader.rs:300:42

---- network::stream_reader::test::read_singlemsg_test stdout ----
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: InvalidChecksum { expected: [55, 14, 242, 191], actual: [190, 97, 184, 39] }', src/network/stream_reader.rs:216:67

---- util::address::tests::test_bip173_350_vectors stdout ----
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Base58(BadByte(48))', src/util/address.rs:707:50

---- util::address::tests::test_qr_string stdout ----
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Base58(BadChecksum(2355552274, 1724156324))', src/util/address.rs:837:46

---- util::base58::tests::test_base58_encode stdout ----
thread 'main' panicked at 'assertion failed: `(left == right)`
  left: `"1PfJpZsjreyVrqeoAfabrRwwjQyoXRZiFv"`,
 right: `"1PfJpZsjreyVrqeoAfabrRwwjQyoSQMmHH"`', src/util/base58.rs:285:9

---- util::bip158::test::test_blockfilters stdout ----
thread 'main' panicked at 'assertion failed: `(left == right)`
  left: `[1, 157, 252, 168]`,
 right: `[1, 125, 64, 192]`', src/util/bip158.rs:573:13

---- util::ecdsa::tests::test_key_derivation stdout ----
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Base58(BadChecksum(3605035416, 1319098282))', src/util/ecdsa.rs:425:95

---- util::merkleblock::tests::merkleblock_construct_from_txids_found stdout ----
thread 'main' panicked at 'assertion failed: `(left == right)`
  left: `cfb0fd226b097dc515a16689a96a3158e4cacf7429162bdcdc2ede0d9d433de8`,
 right: `2fda58e5959b0ee53c5253da9b9f3c0c739422ae04946966991cf55895287552`', src/util/merkleblock.rs:674:9

---- util::merkleblock::tests::merkleblock_construct_from_txids_not_found stdout ----
thread 'main' panicked at 'assertion failed: `(left == right)`
  left: `cfb0fd226b097dc515a16689a96a3158e4cacf7429162bdcdc2ede0d9d433de8`,
 right: `2fda58e5959b0ee53c5253da9b9f3c0c739422ae04946966991cf55895287552`', src/util/merkleblock.rs:707:9

---- util::psbt::tests::bip_vectors::valid_vector_3 stdout ----
thread 'main' panicked at 'assertion failed: `(left == right)`
  left: `f61b1742ca13176464adb3cb66050c00787bb3a4eead37e985f2df1e37718126`,
 right: `c18ae00954e21fe8a5876b5f11e56774cd10836c6b56c5c563b6348874fddf57`', src/util/psbt/mod.rs:696:13


failures:
    blockdata::block::tests::block_test
    blockdata::block::tests::segwit_block_test
    blockdata::block::tests::validate_pow_test
    blockdata::constants::test::bitcoin_genesis_full_block
    blockdata::constants::test::testnet_genesis_full_block
    blockdata::transaction::tests::test_sighash
    network::message::test::deserialize_getaddr_test
    network::message::test::deserialize_partial_message_test
    network::message::test::deserialize_version_test
    network::stream_reader::test::parse_multipartmsg_test
    network::stream_reader::test::read_doublemsgs_test
    network::stream_reader::test::read_multipartmsg_test
    network::stream_reader::test::read_sequencemsg_test
    network::stream_reader::test::read_singlemsg_test
    util::address::tests::test_bip173_350_vectors
    util::address::tests::test_qr_string
    util::base58::tests::test_base58_encode
    util::bip158::test::test_blockfilters
    util::ecdsa::tests::test_key_derivation
    util::merkleblock::tests::merkleblock_construct_from_txids_found
    util::merkleblock::tests::merkleblock_construct_from_txids_not_found
    util::psbt::tests::bip_vectors::valid_vector_3

test result: FAILED. 147 passed; 22 failed; 0 ignored; 0 measured; 0 filtered out; finished in 15.09s

File: src/util/address.rs [hunk 14]
```diff
@@ -838,7 +838,7 @@ mod tests {
             assert_eq!(addr.to_qr_uri(), format!("groestlcoin:{}", el));
         }
 
-        for el in ["bcrt1q2nfxmhd4n3c8834pj72xagvyr9gl57n5r94fsl", "bc1qwqdg6squsna38e46795at95yu9atm8azzmyvckulcc7kytlcckxswvvzej"].iter() {
+        for el in ["bcrt1q2nfxmhd4n3c8834pj72xagvyr9gl57n5r94fsl", "grs1qwqdg6squsna38e46795at95yu9atm8azzmyvckulcc7kytlcckxsphnwkk"].iter() {
```
Comment (gruve-p) [inline src/util/address.rs]: `bcrt1q2nfxmhd4n3c8834pj72xagvyr9gl57n5r94fsl` is bitcoin regtest address

File: src/consensus/params.rs [hunk 15]
```diff
@@ -46,7 +46,7 @@ const MAX_BITS_REGTEST: Uint256 = Uint256([
     0x0000000000000000u64,
     0x0000000000000000u64,
     0x0000000000000000u64,
-    0x7fffff0000000000u64,
+    0x00ffffffffffffffu64,
```
Comment (HashEngineering) [inline src/consensus/params.rs:49]: This is the number in `SigNetParams`
`00000377ae000000000000000000000000000000000000000000000000000000`

Comment (HashEngineering) [reply src/consensus/params.rs:49]: `00ffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff` is RegTest

Comment (HashEngineering) [reply src/consensus/params.rs:49]: Nevermind, looking at the wrong code.

File: contrib/test.sh (hunk 1 above)
Comment (gruve-p) [reply contrib/test.sh]: Fixed

File: embedded/README.md (hunk 2 above)
Comment (gruve-p) [reply embedded/README.md]: Fixed

File: embedded/src/main.rs (hunk 3 above)
Comment (gruve-p) [reply embedded/src/main.rs]: fixed

File: embedded/src/main.rs (hunk 4 above)
Comment (gruve-p) [reply embedded/src/main.rs]: fixed

File: examples/bip32.rs (hunk 5 above)
Comment (gruve-p) [reply examples/bip32.rs]: fixed

File: src/util/address.rs (hunk 9 above)
Comment (gruve-p) [reply src/util/address.rs]: fixed

File: src/util/base58.rs (hunk 10 above)
Comment (gruve-p) [reply src/util/base58.rs]: fixed

File: src/util/ecdsa.rs (hunk 11 above)
Comment (gruve-p) [reply src/util/ecdsa.rs]: fixed

File: src/util/ecdsa.rs (hunk 12 above)
Comment (gruve-p) [reply src/util/ecdsa.rs]: fixed

File: src/util/ecdsa.rs (hunk 13 above)
Comment (gruve-p) [reply src/util/ecdsa.rs]: fixed

File: src/util/address.rs (hunk 14 above)
Comment (gruve-p) [reply src/util/address.rs]: fixed

**Review by gruve-p — APPROVED** [review]

**Pull request merged**